src/executor.ts

```diff
@@ -362,9 +362,16 @@ class ScriptExecution {
     // Note we use Promise.allSettled instead of Promise.all so that we can
     // collect all errors, instead of just the first one.
     const dependencyResults = await Promise.allSettled(
-      this.#script.dependencies.map((dependency) =>
-        this.#executor.execute(dependency)
-      )
+      this.#script.dependencies.map((dependency) => {
```

Comment:
It sort of feels like we should be able to enforce that an `Executor` only takes a config that is guaranteed to be valid from the `Analyzer`, so that we don't have to check this during execution?